app/controllers/reviews_controller.rb

```diff
@@ -1,38 +1,65 @@
 class ReviewsController < ApplicationController
+  before_action :set_paging, only: %w[index of_product]  
 
-  DEFAULT_TAGS = ['default']
-
+  # GET /
   def index
-    if params[:shop_id].present? && Shop.where("id = #{params[:shop_id]}").exists?
-      params[:per_page] ||= 10
-      offset = params[:page].to_i * params[:per_page]
-
-      @data = []
-      products = Product.where("shop_id = #{params[:shop_id]}").sort_by(&:created_at)[offset..(offset + params[:per_page])]
-      products.each do |product|
-        reviews = product.reviews.sort_by(&:created_at)[offset..(offset + params[:per_page])]
-        @data << { product: product, reviews: reviews }
-      end
+    # Get store listing for filter
+    @shops = Shop.order(:name).map { |s| [s.name, s.id] }
```

Comment:
This would be simpler and faster for DB and the web server:
```suggestion
    @shops = Shop.order(:name).pluck(:name, :id)
```